perfkitbenchmarker/scripts/step_load_driver/launch_mysql_service.py

```diff
@@ -0,0 +1,258 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""launch_mysql_service is a wrapper class for mysql_service_benchmark.
+It will call mysql_service_benchmark with varying thread counts depending on
+thread_count_list flag values.
+
+If a run_uri flag is provided, the program assumes that the PKB instance
+has been provisioned and prepared, and will only execute the run phase of PKB.
+If no run_uri is given, the program will execute the provision and prepare
+phase followed by consecutive run phases and then the cleanup and teardown
+phase. Below are examples of possible ways to call this class.
+
+Possible call:
+./launch_mysql_service --thread_count_list=[1,2,5] --sysbench_run_seconds=20
+./launch_mysql_service --run_uri=12e4s6t8 --thread_count_list=[1,2,4,8,10]
+
+All requirements for mysql_service_benchmark still apply. See
+perkfitbenchmarker.linuxbenchmarks.mysql_service_benchmark for more details.
+
+Any additional flags not specifically outlined in the flags below can be added
+as a list of strings under the 'additional_flags' flag. For example, a possible
+call with additional flags could be:
+./launch_mysql_service --run_uri=2u2u2u3i
+  --additional_flags=['--storage_size=100', '--cloud=GCP'].
+"""
+
+import datetime
+import logging
+import re
+import shlex
+import subprocess
+import sys
+import time
+import gflags
+
+# GLOBAL STRINGS
+SYSBENCH_RUN_SECONDS = 'sysbench_run_seconds'
+SYSBENCH_THREAD_COUNT = 'sysbench_thread_count'
+SYSBENCH_REPORT_INTERVAL = 'sysbench_report_interval'
+STORAGE_BUCKET = 'storage_bucket'
+THREAD_COUNT_LIST = 'thread_count_list'
+RUN_URI = 'run_uri'
+DEFAULT_RUN_TIME = 60
+STDOUT = 'STDOUT'
+STDERR = 'STDERR'
+DATETIME_FORMAT = '{:%m_%d_%Y_%H_%M_}'
+URI_REGEX = r'run_uri=([a-z0-9]{8})'
+ADDITIONAL_FLAGS = 'additional_flags'
+
+MAX_SLEEP_ITER = 24  # max wait time for a run is max_sleep_iter x sleep_time
+SLEEP_TIME = 1800  # seconds
+
+# FLAG STRINGS
+PKB = './pkb.py --benchmarks=mysql_service'
+STAGE_FLAG = ' --run_stage='
+URI_FLAG = ' --run_uri='
+THREAD_FLAG = ' --sysbench_thread_count='
+RUN_TIME = ' --sysbench_run_seconds='
+STORAGE = ' --cloud_storage_bucket='
+PROVISION = 'provision'
+PREPARE = 'prepare'
+RUN = 'run'
+CLEANUP = 'cleanup'
+TEARDOWN = 'teardown'
+
+FLAGS = gflags.FLAGS
+gflags.DEFINE_integer(SYSBENCH_RUN_SECONDS, 480,
+                      'The duration of the actual run in which results are '
+                      'collected, in seconds.')
+gflags.DEFINE_list(THREAD_COUNT_LIST, [1, 2, 5, 8],
+                   'The number of test threads on the client side.')
+gflags.DEFINE_integer(SYSBENCH_REPORT_INTERVAL, 2,
+                      'The interval, in seconds, we ask sysbench to report '
+                      'results.')
+gflags.DEFINE_string(RUN_URI, None,
+                     'Run identifier, if provided, only run phase '
+                     'will be completed.')
+gflags.DEFINE_string(STORAGE_BUCKET, None,
```

Comment:
same as above, should we just pass through storage_bucket flag as additional flags?